source/Mlos.Python/mlos/Spaces/Hypergrid.py

```diff
@@ -50,13 +59,73 @@ def get_dimensions_for_point(self, point, return_join_dimensions=True):
     def random(self, point=None):
         raise NotImplementedError("All subclasses must implement this.")
 
+    @trace()
     def random_dataframe(self, num_samples):
         config_dicts = [
             {dim_name: value for dim_name, value in self.random()}
             for _ in range(num_samples)
         ]
         return pd.DataFrame(config_dicts)
 
+    @trace()
+    def filter_out_invalid_rows(self, original_dataframe: pd.DataFrame, exclude_extra_columns=True) -> pd.DataFrame:
+        """ Returns a dataframe containing only valid rows from the dataframe.
+
+        Valid rows are rows with no NaNs and with values for all dimensions in the required ranges.
+
+        If there are additional columns, they will be preserved.

```

Comment:
the default is ``exclude_extra_columns=True`` which would mean they are not preserved by default, right?

Reply:
Good point.